ConfigSpace/hyperparameters.pyx

```diff
@@ -877,6 +998,327 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
+                                           default_value=int(np.round(self.default_value, 0)),
+                                           q=q_int, log=self.log)
+
+    def is_legal(self, value: Union[float]) -> bool:
+        if not (isinstance(value, float) or isinstance(value, int)):
+            return False
+        elif self.upper >= value >= self.lower:
+            return True
+        else:
+            return False
+
+    cpdef bint is_legal_vector(self, DTYPE_t value):
+        if self.upper >= value >= self.lower:
+            return True
+        else:
+            return False
+
+    def _sample(self, rs: np.random.RandomState, size: Optional[int] = None
+                ) -> Union[np.ndarray, float]:
+
+        if (self.lower == None) or (self.upper == None):
+            return ValueError('The beta hyperparameter must have both bounds specified.')
+        else:
+            alpha = self.alpha
+            beta = self.beta
+            lower = self._lower
+            upper = self._upper
+            return spbeta(alpha, beta, loc=lower, scale=upper-lower).rvs(size=size, random_state=rs)
+
+    cpdef np.ndarray _transform_vector(self, np.ndarray vector):
+        if np.isnan(vector).any():
+            raise ValueError('Vector %s contains NaN\'s' % vector)
+        if self.log:
+            vector = np.exp(vector)
+        if self.q is not None:
+            vector = np.rint(vector / self.q) * self.q
+        return vector
+
+    cpdef double _transform_scalar(self, double scalar):
+        if scalar != scalar:
+            raise ValueError('Number %s is NaN' % scalar)
+        if self.log:
+            scalar = math.exp(scalar)
+        if self.q is not None:
+            scalar = round(scalar / self.q) * self.q
+        return scalar
+        
+    def _inverse_transform(self, vector: Optional[np.ndarray]) -> Union[float, np.ndarray]:
+        if vector is None:
+            return np.NaN
+
+        if self.log:
+            vector = np.log(vector)
+        return vector
+
+    def get_neighbors(
+        self,
+        value: Any,
+        rs: np.random.RandomState,
+        number: int = 4,
+        transform: bool = False,
+        std: float = 0.2
+    ) -> List[float]:
+        neighbors = []  # type: List[float]
+        norm_std = std * (self._upper - self._lower)
+        while len(neighbors) < number:
+            neighbor = rs.normal(value, norm_std)  # type: float
+            if neighbor < self._lower or neighbor > self._upper:
+                continue
+            if transform:
+                neighbors.append(self._transform(neighbor))
+            else:
+                neighbors.append(neighbor)
+        return neighbors
+
+    def _pdf(self, vector: np.ndarray) -> np.ndarray:
+        alpha = self.alpha
+        beta = self.beta
+        if self.lower == None:
+            return ValueError('The beta hyperparameter must have both bounds specified.')
```

Comment:
Same here for error checking if it's non-optional

Reply:
And fixed